src/test/java/com/ericsson/ei/frontend/BackendInstancesUtilsTest.java

```diff
@@ -60,44 +55,115 @@
     public void before() throws IOException {
         instance = new JsonParser().parse(new FileReader(BACKEND_INSTANCE_FILE_PATH)).getAsJsonObject();
         instances = new JsonParser().parse(new FileReader(BACKEND_INSTANCES_FILE_PATH)).getAsJsonArray();
-        utils.setInstances(instances);
-    }
 
-    @Test
-    public void testSetBackendProperties() {
-        BackEndInformation information = new BackEndInformation(NAME_VAL, HOST_VAL, String.valueOf(PORT_VAL), PATH_VAL, HTTPS_VAL, ACTIVE_VAL);
-        utils.setBackEndProperties(information);
-        assertEquals(NAME_VAL, info.getName());
-        assertEquals(HOST_VAL, info.getHost());
-        assertEquals(PORT_VAL, Integer.parseInt(info.getPort()));
-        assertEquals(PATH_VAL, info.getPath());
-        assertEquals(HTTPS_VAL, info.isUseSecureHttpBackend());
-        assertEquals(ACTIVE_VAL, info.isActive());
+        utils.getDefaultBackendInformation().setHost(null);
     }
 
     @Test
     public void testCheckIfInstanceAlreadyExistTrue() {
+        when(fileUtils.getInstancesFromFile()).thenReturn(instances);
         boolean result = utils.checkIfInstanceAlreadyExist(instance);
-        assertEquals(true, result);
+        assertEquals("checkIfInstanceAlreadyExist should return 'true', but returned '" + result + "'.",
+                true, result);
     }
 
     @Test
     public void testCheckIfInstanceAlreadyExistFalse() {
+        when(fileUtils.getInstancesFromFile()).thenReturn(instances);
+
         JsonObject newInstance = instance.getAsJsonObject();
         newInstance.addProperty("host", "newHost");
         newInstance.addProperty("port", newInstance.get("port").getAsInt() + 1);
         newInstance.addProperty("path", "newPath");
         boolean result = utils.checkIfInstanceAlreadyExist(newInstance);
-        assertEquals(false, result);
+
+        assertEquals("checkIfInstanceAlreadyExist should return 'false', but returned '" + result + "'.",
+                false, result);
+    }
+
+    @Test
+    public void testCheckIfInstanceNameAlreadyExistTrue() {
+        when(fileUtils.getInstancesFromFile()).thenReturn(instances);
+
+        JsonObject newInstance = instance.getAsJsonObject();
```

Comment:
this is strange. If you add an instance with the same name but different port it fails on checking if same name exists.

But if you add a new instance with same name and different port as above it will return that the instance is not already existing. If I misunderstood the test please make it more clear.

Reply:
A name should always be unique, the function tests names nothing else. The test add a random port to show that id still does not accept it.